### DISCUSSION TITLE
TEX_FORMAT_RGBA8_SNORM for Normal Maps

### DISCUSSION THREAD

**Discussion by RobertMtx** [DiligentGraphics/DiligentEngine#225] [General]
When I attempt to load a normal map with format TEX_FORMAT_RGBA8_SNORM, I get strange values that seem to be out of range.

I tested this by loading the same normal map with format TEX_FORMAT_RGBA8_UNORM, and directly rendering the R component to the screen with no lighting applied. What I see is identical to what I see when I do the same thing in Blender's shader. However, when I try loading the normal map with TEX_FORMAT_RGBA8_SNORM, then render the value `(normals.x + 1.0) * 0.5`, I do not get the same result. The value range appears to be much greater. I've tried normalizing before using it, and it looks the same, so I'm assuming it is already normalized. I've tried searching general forums to see if there's something obvious I'm missing, but can't seem to find anything. Am I misunderstanding the format of this type of texture?

Any advice on what I may be doing wrong here? I can post images of the results, if needed. Thanks!

**Comment by TheMostDiligent** [comment]
I've also had some issues with snorm textures for normal maps, so I never use them - always unorm textures with explicit conversion to signed range

**Comment by RobertMtx** [comment]
It's strange, because the z (blue) value looks nearly correct, but x & y look as though the contrast has been cranked way up. If I render the x or y values as max(0,value) or max(0,-value), I get pure black and white pixels, with very few greys. It looks as though the values are being scaled way too far.

Thanks for confirming!
